dms/models/dms_file.py

```diff
@@ -638,7 +638,8 @@ def _inverse_content(self):
 
     def _create_model_attachment(self, vals):
         res_vals = vals.copy()
-        directory = self.env["dms.directory"].sudo().browse(res_vals["directory_id"])
+        directory_id = res_vals.get("directory_id", self.env.context.get("active_id"))
+        directory = directory_id and self.env["dms.directory"].browse(directory_id)
         if directory and directory.res_model and directory.res_id:
```

Comment:
You can safely do this:

```suggestion
        directory = self.env["dms.directory"].browse(directory_id)
        if directory.res_model and directory.res_id:
```

It is safe to do it, because in case `directory_id` is `False` or `None`, then `.browse()` will return an empty recordset, which itself returns always falsy values when accessing its fields.